frontend/src/lib/components/VersionChecker/versionCheckerLogic.ts

```diff
@@ -0,0 +1,123 @@
+    }),
+    loaders({
+        availableVersions: [
+            null as SDKVersion[] | null,
+            {
+                loadAvailableVersions: async () => {
+                    return await fetch('https://api.github.com/repos/posthog/posthog-js/tags')
+                        .then((r) => r.json())
+                        .then((r) => r.map((x: any) => ({ version: x.name.replace('v', '') })))
+                },
+            },
+        ],
+        usedVersions: [
+            null as SDKVersion[] | null,
+            {
+                loadUsedVersions: async () => {
+                    const query: HogQLQuery = {
+                        kind: NodeKind.HogQLQuery,
+                        query: `SELECT properties.$lib_version AS lib_version, max(timestamp) AS latest_timestamp, count(lib_version) as count
+                                FROM events
+                                WHERE timestamp >= now() - INTERVAL 1 DAY 
+                                AND timestamp <= now()
+                                AND properties.$lib = 'web'
+                                GROUP BY lib_version
+                                ORDER BY latest_timestamp DESC
+                                limit 10`,
+                    }
+
+                    const res = await api.query(query)
+
+                    return (
+                        res.results?.map((x) => ({
+                            version: x[0],
+                            timestamp: x[1],
+                        })) ?? null
+                    )
+                },
+            },
+        ],
+    }),
+
+    reducers({
+        lastCheckTimestamp: [
+            0,
+            { persist: true },
+            {
+                loadUsedVersionsSuccess: () => Date.now(),
+            },
+        ],
+        versionWarning: [
+            null as SDKVersionWarning | null,
+            { persist: true },
+            {
+                setVersionWarning: (_, { versionWarning }) => versionWarning,
+            },
+        ],
+    }),
+
+    sharedListeners(({ values, actions }) => ({
+        checkForVersionWarning: () => {
+            if (!values.availableVersions?.length || !values.usedVersions?.length) {
+                return
+            }
+
+            const latestVersion = values.availableVersions[0].version
+            const currentVersion = values.usedVersions[0].version
+
+            if (latestVersion === currentVersion) {
+                actions.setVersionWarning(null)
+                return
+            }
+
+            let diff = values.availableVersions.findIndex((v) => v.version === currentVersion)
+            diff = diff === -1 ? values.availableVersions.length : diff
+
+            const warning: SDKVersionWarning = {
+                currentVersion,
+                latestVersion,
+                diff,
+                level: diff > 20 ? 'error' : diff > 10 ? 'warning' : 'info',
```

Comment:
Didn't think about it until the demo just now but I wonder if a better metric for severity would be patch, minor and major updates? Might not be super easy to calculate but just a thought